### PR TITLE
Remove some usages of OrderedDict

### PR DESCRIPTION
**Pull request by timhoffm**
... and simplify related code.

### CODE REVIEW

File: lib/matplotlib/tests/test_style.py [hunk 1]
```diff
@@ -138,10 +137,9 @@ def test_context_with_union_of_dict_and_namedstyle():
 def test_context_with_badparam():
     original_value = 'gray'
     other_value = 'blue'
-    d = OrderedDict([(PARAM, original_value), ('badparam', None)])
     with style.context({PARAM: other_value}):
         assert mpl.rcParams[PARAM] == other_value
-        x = style.context([d])
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_style.py:144]: the additional wrapping in a list is unnecessary here.

File: lib/matplotlib/backends/backend_pdf.py [hunk 2]
```diff
@@ -683,14 +682,14 @@ def __init__(self, filename, metadata=None):
         self._soft_mask_seq = (Name(f'SM{i}') for i in itertools.count(1))
         self._soft_mask_groups = []
         # reproducible writeHatches needs an ordered dict:
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_pdf.py:684]: dunno if you want to update the comment here and below

Comment (timhoffm) [reply lib/matplotlib/backends/backend_pdf.py]: We can just remove it.

**Review by anntzer — APPROVED** [review]

**Comment by anntzer** [comment]
(modulo ci)

File: lib/matplotlib/tests/test_style.py [hunk 3]
```diff
@@ -138,10 +137,9 @@ def test_context_with_union_of_dict_and_namedstyle():
...
+        x = style.context({PARAM: original_value, 'badparam': None})
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_style.py:142]: Do you need `x`, or can it just in the `with`?

Comment (timhoffm) [reply lib/matplotlib/tests/test_style.py:142]: I suppose this was written to check that the error occurs when entering the context, and not when creating it.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**